### PR TITLE
[NT-228, NT-376] Cancel Pledge Error Handling

### PR DESCRIPTION
**Pull request by ifbarrera**
# 📲 What

Adds error handling via the message banner for the cancel pledge flow. This PR also adds the missing accessibility label on the cancellation reason text field, and updates the disabled and highlighted colors on the "Yes, cancel it" button.

# 🤔 Why

To let users know when cancelling their pledge has failed.

# 🛠 How

- adds `MessageBannerPresenting` conformance to `CancelPledgeViewController`
- hooks the banner up to the `cancelPledgeError` output
- adds an accessibility label to the `cancellationReasonTextField`
- updates the `.disabled` and `.highlighted` colors on the `redButtonStyle`

# 👀 See

tTUywwXmxU

# ♿️ Accessibility 

- [ ] The cancellation reason text field should read "Cancellation reason, Tell us why, optional" with voice over enabled

# ✅ Acceptance criteria

- [x] Navigate to a backing for a project that was not created on staging (any backing really). Navigate to the cancel pledge screen and try to pledge by tapping "Yes, cancel it". The button should become disabled and you should see an error banner animate in.

### CODE REVIEW

File: Kickstarter-iOS/Locales/ja.lproj/Localizable.strings [hunk 1]
```diff
@@ -60,6 +60,7 @@
 "Cancel" = "キャンセル";
 "Cancel_pledge" = "プレッジを取り消す";
 "Cancel_your_pledge" = "プレッジを取り消す";
+"Cancellation_reason" = "Motivo de cancelación";
```
Comment (cdolm92) [inline Kickstarter-iOS/Locales/ja.lproj/Localizable.strings]: Why are the Japanese strings in Spanish? 🤔

File: Kickstarter-iOS/Views/Controllers/CancelPledgeViewController.swift [hunk 2]
```diff
@@ -122,6 +125,7 @@ final class CancelPledgeViewController: UIViewController {
 
     _ = self.cancellationReasonTextField
       |> cancellationReasonTextFieldStyle
+      |> UITextField.lens.accessibilityLabel %~ { _ in Strings.Cancellation_reason() }
       |> UITextField.lens.placeholder %~ { _ in Strings.Tell_us_why_optional() }
```
Comment (cdolm92) [inline Kickstarter-iOS/Views/Controllers/CancelPledgeViewController.swift]: Can we get rid of the lens and do ` |> \.accessibilityLabel %~ { _ in Strings.Cancellation_reason() }` and  ` |> \.placeholder %~ { _ in Strings.Tell_us_why_optional() }` here?

File: Library/Strings.swift [hunk 3]
```diff
@@ -1004,6 +1004,23 @@ contributeurs"
...
     )
   }
+  /**
+   "Cancellation reason"
+
+   - **en**: "Cancellation reason"
+   - **de**: "Grund der Annullierung"
+   - **es**: "Motivo de cancelación"
+   - **fr**: "Motif d'annulation"
+   - **ja**: "Motivo de cancelación"
```
Comment (cdolm92) [inline Library/Strings.swift]: Here too. Something went wrong with the Japanese translations.

File: Kickstarter-iOS/Locales/ja.lproj/Localizable.strings (hunk 1 above)
Comment (ifbarrera) [reply Kickstarter-iOS/Locales/ja.lproj/Localizable.strings]: Good question...

**Pull request merged**